src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -0,0 +1,696 @@
+            }
+            if (stderrPipeFds != null)
+            {
+                Interop.libc.close(stderrPipeFds[1]); // close copy of child's stderr
+                _standardError = new StreamReader(
+                    OpenStream(stderrPipeFds[0], FileAccess.Read),
+                    startInfo.StandardErrorEncoding ?? Encoding.UTF8, true, StreamBufferSize);
+            }
+
+            return true;
+        }
+
+        // -----------------------------
+        // ---- PAL layer ends here ----
+        // -----------------------------
+
+        /// <summary>Finalizable holder for the underlying shared wait state object.</summary>
+        private ProcessWaitState.Holder _waitStateHolder;
+
+        /// <summary>Size to use for redirect streams and stream readers/writers.</summary>
+        private const int StreamBufferSize = 4096;
+
+        /// <summary>Resolves a path to the filename passed to ProcessStartInfo.</summary>
+        /// <param name="filename">The filename.</param>
+        /// <returns>The resolved path.</returns>
+        private static string ResolvePath(string filename)
+        {
+            // Follow the same resolution that Windows uses with CreateProcess:
+            // 1. First try the exact path provided
+            // 2. Then try the file relative to the executable directory
+            // 3. Then try the file relative to the current directory
+            // 4. then try the file in each of the directories specified in PATH
+            // Windows does additional Windows-specific steps between 3 and 4,
+            // and we ignore those here.
+
+            // If the filename is a complete path, use it, regardless of whether it exists.
+            if (Path.IsPathRooted(filename))
+            {
+                // In this case, it doesn't matter whether the file exists or not;
+                // it's what the caller asked for, so it's what they'll get
+                return filename;
+            }
+
+            // Then check the executable's directory
+            string path = GetExePath();
+            if (path != null)
+            {
+                try
+                {
+                    path = Path.Combine(Path.GetDirectoryName(path), filename);
+                    if (File.Exists(path))
+                    {
+                        return path;
+                    }
+                }
+                catch (ArgumentException) { } // ignore any errors in data that may come from the exe path
+            }
+
+            // Then check the current directory
+            path = Path.Combine(Directory.GetCurrentDirectory(), filename);
+            if (File.Exists(path))
+            {
+                return path;
+            }
+
+            // Then check each directory listed in the PATH environment variables
+            string pathEnvVar = Environment.GetEnvironmentVariable("PATH");
+            if (pathEnvVar != null)
```

Comment:
Yup, I've separated it out into a reusable helper in Common\src\System\IO\StringParser.cs.